src/openms/include/OpenMS/KERNEL/SpectrumRangeManager.h

```diff
@@ -0,0 +1,156 @@
+    This class extends the basic RangeManager to provide separate range tracking for different MS levels
+    (MS1, MS2, etc.). It manages four types of ranges:
+    - m/z (mass-to-charge ratio)
+    - intensity
+    - retention time (RT)
+    - ion mobility
+    
+    A global range is tracked for all MS levels, and additional ranges are maintained for each specific MS level.
+    This allows for efficient querying of ranges for specific MS levels, which is useful for visualization,
+    filtering, and processing operations that need to work with specific MS levels.
+    
+    The class inherits from RangeManager and adds MS level-specific functionality. The base RangeManager
+    functionality is used for the global ranges, while a map of MS levels to RangeManagers is used for
+    the MS level-specific ranges.
+
+    @see RangeManager
+    @see MSSpectrum
+    @see ChromatogramRangeManager
+    @see MSExperiment
+    @ingroup Kernel
+  */
+  class OPENMS_DLLAPI SpectrumRangeManager : public RangeManager<RangeMZ, RangeIntensity, RangeMobility, RangeRT>
+  {
+  public:
+    /// Base type
+    using BaseType = RangeManager<RangeMZ, RangeIntensity, RangeMobility, RangeRT>;
+    
+    /// Default constructor
+    SpectrumRangeManager() = default;
+    
+    /// Copy constructor
+    SpectrumRangeManager(const SpectrumRangeManager& source) = default;
+    
+    /// Move constructor
+    SpectrumRangeManager(SpectrumRangeManager&& source) = default;
+    
+    /// Assignment operator
+    SpectrumRangeManager& operator=(const SpectrumRangeManager& source) = default;
+    
+    /// Move assignment operator
+    SpectrumRangeManager& operator=(SpectrumRangeManager&& source) = default;
+    
+    /// Destructor
+    ~SpectrumRangeManager() = default;
+    
+    /**
+      @brief Clears all ranges (global and MS level-specific)
+    */
+    void clearRanges()
+    {
+      BaseType::clearRanges();
+      ms_level_ranges_.clear();
+    }
+    
+    /**
+      @brief Extends the ranges with the ranges of another range manager
+      
+      @param other The other range manager to extend from
+      @param ms_level The MS level for which to extend the ranges (0 for global ranges)
+    */
+    void extend(const BaseType& other, UInt ms_level = 0)
+    {
+      ms_level == 0 ? BaseType::extend(other) : ms_level_ranges_[ms_level].extend(other);
+    }
+    
+    /**
+      @brief Gets the ranges for a specific MS level
+      
+      @param ms_level The MS level for which to retrieve the ranges
+      @return The ranges for the specified MS level
+      @throw Exception::InvalidValue if no ranges exist for the specified MS level
+    */
+    const BaseType& byMSLevel(UInt ms_level = 0) const
+    {
+      if (ms_level_ranges_.find(ms_level) == ms_level_ranges_.end())
+      {
+        throw Exception::InvalidValue(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, "No ranges for this MS level", String(ms_level));
+      }
+      return ms_level_ranges_.at(ms_level);
```

Comment:
just out of good habit (avoid double lookup):
```suggestion
      if (auto it = ms_level_ranges_.find(ms_level); it != ms_level_ranges_.end())
      {
        return it->second;
      } 
      throw Exception::InvalidValue(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, "No ranges for this MS level", String(ms_level));
```